src/related-items-and-comparisons/RelatedItemsList.jsx

```diff
@@ -1,12 +1,62 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import CardItem from './CardItem.jsx';
 
 const RelatedItemsList = (props) => {
-  return (
-    <div className="related-items-list">
-      {props.products.map((product) => <CardItem key={product.id} product={product} />)}
-    </div>
-  );
+  const allProducts = props.products;
+  const products = props.products;
+  const [currList, setCurrList] = useState(products);
+  const [currIndex, setCurrIndex] = useState(0);
+
+  // "useEffect is componentDidMount, componentDidUpdate, and componentWillUnmount combined"
+
+  const next = () => {
+    setCurrIndex(currIndex + 1);
+    setCurrList(products.slice(currIndex + 1)); //bandaid fix, still not sure why its not updating the array without the +1 here
+  };
+
+  const prev = () => {
+    setCurrIndex(currIndex - 1);
+    setCurrList(products.slice(currIndex - 1));
+  };
+
+  if (currIndex === products.length - 1) {
```

Comment:
Instead of all this DRY violation, create a left arrow component and right arrow component and set them to null if they shouldn't render. [Here's an example of that.](https://github.com/Team-Asteraceae/project-catwalk/blob/67c0f6f73cdbcd94093b6b760a99ba811aeaa907/src/overview/ImageGallery.jsx#L101)

Reply:
Just re-implemented using your example. That solution is sick thanks for the guidance!